cli/src/pcluster/models/cluster.py

```diff
@@ -909,14 +909,18 @@ def update(
             LOGGER.critical(e)
             raise _cluster_error_mapper(e, f"Cluster update failed.\n{e}")
 
-    def _add_version_tag(self):
+    def _add_pcluster_tags(self):
         """Add version tag to the stack."""
         if self.config.tags is None:
             self.config.tags = []
         # Remove PCLUSTER_VERSION_TAG if already exists
-        self.config.tags = [tag for tag in self.config.tags if tag.key != PCLUSTER_VERSION_TAG]
+        tag_types = {PCLUSTER_CLUSTER_NAME_TAG, PCLUSTER_VERSION_TAG}
```

Comment:
Instead of appending the list twice, we can do something like:
```
cluster_stack_tags = {PCLUSTER_CLUSTER_NAME_TAG: self.name, PCLUSTER_VERSION_TAG: get_installed_version()}
# Remove PCLUSTER_VERSION_TAG and ... if already exists
self.config.tags = [tag for tag in self.config.tags if tag.key not in cluster_stack_tags]
# Add PCLUSTER_VERSION_TAG and ...
self.config.tags += [{"Key": key, "Value": value} for key, value in cluster_stack_tags.items()]
```

Reply:
Good idea. Updated!